File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/Application/Omega/AccountNumberAssociation.cs
namespace Psi.Data.Models.ClientConfigurationModels.Application.Omega
{
    public class AccountNumberAssociation : SettingsBaseHelper
    {

        public AccountNumberAssociation(ISettingsBase settingsBase) : base(settingsBase)
        {
        }

        /// <summary>
        /// Account number association enabled as bool
        /// </summary>
        [SettingKey("Omega.Features.AccountNumberAssociationEnabled")]
        public bool Enabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        /// <summary>
        /// If true, then deactivate external scheduled transfers during account number re-association, rather than re-associating them.
        /// </summary>
        [SettingKey("Omega.Features.AccountNumberAssociation.DeactivateExternalScheduledTransfers")]
        public bool DeactivateExternalScheduledTransfers
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }
    }
}

File: infrastructure/models/ClientConfigurationModels/Application/Omega/AccountNumberAssociation.ts
import { Setting, ISettingsGroup, ISettingsMetadata } from '@models/base/types';
export interface AccountNumberAssociationConfig {
    Enabled: boolean;
    DeactivateExternalScheduledTransfers: boolean;
}

export class AccountNumberAssociation implements ISettingsGroup {
    static readonly metadata: ISettingsMetadata = {
        groupName: 'AccountNumberAssociation'
    };


            private _enabled: boolean;
            get enabled(): boolean {
                return this._enabled;
            }
            set enabled(value: boolean) {
                this._enabled = value;
            }

            private _deactivateExternalScheduledTransfers: boolean;
            get deactivateExternalScheduledTransfers(): boolean {
                return this._deactivateExternalScheduledTransfers;
            }
            set deactivateExternalScheduledTransfers(value: boolean) {
                this._deactivateExternalScheduledTransfers = value;
            }


    constructor() {}


        toSettings(): Setting[] {
            return [
                { key: "AccountNumberAssociation.Enabled", value: this._enabled, dataType: 'boolean', label: "Enabled" },
                { key: "AccountNumberAssociation.DeactivateExternalScheduledTransfers", value: this._deactivateExternalScheduledTransfers, dataType: 'boolean', label: "Deactivate External Scheduled Transfers" },
            ];
        }

}